### ISSUE TITLE
Type definitions for Handsontable.GridSettings.renderer and Handsontable.GridSettings.validator are not correct

### ISSUE THREAD

**Issue by IRCraziestTaxi** [handsontable/handsontable#5004]
### Description
The type definitions for Handsontable.GridSettings.renderer and Handsontable.GridSettings.validator are not correct.  I get typescript errors when trying to pass the appropriate arguments into the methods.

### Steps to reproduce
1. Install handsontable 0.35
2. Attempt to define a column's renderer as:

```ts
renderer: (instance, td, row, col, prop, value, cellProperties) {
}
```

3. Attempt to define a column's validator as:

```ts
validator: (value, callback: (condition: boolean) => void) => {
}
```

Type errors are produced.  In `handsontable.d.ts`, within the `GridSettings` interface, those two functions should contain at least the following types:

```ts
renderer?: (
            instance: Core,
            td: HTMLTableCellElement,
            row: number,
            col: number,
            prop: string,
            value: any,
            cellProperties: any
) => void;

validator?: (value: any, callback: (condition: boolean) => void) => void;
```

But, instead, they are typed as follows:

```ts
renderer?: string | (() => void);

validator?: (() => void) | RegExp;
```

### Your environment
* Handsontable version: 0.35

**Comment by NoNameProvided** [comment]
I experience the same issue.

**Comment by IRCraziestTaxi** [comment]
Additional note: This is still true in handsontable v2.0.

**Comment by AMBudnik** [comment]
related https://github.com/handsontable/handsontable/issues/5091

**Comment by swistach** [comment]
Hi @IRCraziestTaxi,
Thank you for your sharing. I marked this issue as a bug.

**Comment by aaronbeall** [comment]
This seems to be fixed, `renderer` and `validator` are now defined in `GridSettings` as:
```
renderer?: string | renderers.Base;
validator?: (value: any, callback: (valid: boolean) => void) => void | RegExp;
```
And `renderers.Base` is:
```
interface Base {
  (instance: _Handsontable.Core, TD: HTMLElement, row: number, col: number, prop: string | number, value: any, cellProperties: GridSettings): HTMLElement;
}
```
Does that resolve this issue, @IRCraziestTaxi ?

**Comment by AMBudnik** [comment]
Hey @IRCraziestTaxi @NoNameProvided 

thanks to @aaronbeall and our dev team we've improved the definition file. Just released it in v `7.0.0` of Handsontable.
